Fixtures/DependencyResolution/External/XCFrameworkExternal/Foo/Foo.swift

```diff
@@ -0,0 +1,3 @@
+public func foo() {
```

Comment:
Is this function called from anywhere? Can this file stay empty and the test still valid that way?

Reply:
whoops, sorry on the title change, I had not realized you changed it when I changed the title.

This function is not called from anywhere, I was copying over the fixture behavior of `XCFramework` which had a `Foo` target with the same code, I can actually just go ahead and delete this `Foo` Swift target if you'd like, and only build the `Baz` target, since the underlying problem is with the `Baz` C++ target.

Reply:
Yes, we should make new tests as clean as possible. But as I said, we may not need a fixture a tool if you can construct build planning inputs like existing build planning tests already do.

Reply:
I went ahead and removed the `Foo` Swift target from the new `XCFrameworkExternal` fixture, this fixture (if we are keeping it) now only has a single `Baz` C++ target, and I went ahead and just reconfirmed this error is still reproducible in this fixture when running `swift build` from the `main` branch.